File: app/src/main/java/com/example/bookworm/bottomMenu/Feed/comments/Comment.java
package com.example.bookworm.bottomMenu.Feed.comments;

import com.google.firebase.firestore.Exclude;

import java.io.Serializable;
import java.text.SimpleDateFormat;
import java.util.Map;

public class Comment implements Serializable {
    //댓글 ID
    private String CommentID;
    //유저 정보
    private String userToken;
    //댓글 내용
    private String contents;
    //생성된 시각
    private String madeDate;

    @Exclude
    private int position; //댓글 삭제시 리사이클러뷰의 포지션을 가져올때 사용

    public Comment() {

    }

    public void getData(String userToken, String contents, Long madeDate) {
        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
        this.CommentID = madeDate + "_" + userToken;
        this.userToken = userToken;
        this.contents = contents;
        this.madeDate = dateFormat.format(madeDate);
    }

    public String getCommentID() {
        return CommentID;
    }

    public void setData(Map data) {
        this.CommentID = (String) data.get("commentID");
        this.contents = (String) data.get("contents");
        this.madeDate = (String) data.get("madeDate");
        this.userToken = (String) data.get("userToken");
    }

    public String getUserToken() {
        return userToken;
    }

    public String getContents() {
        return contents;
    }

    public String getMadeDate() {
        return madeDate;
    }

    public int getPosition() {
        return position;
    }

    public void setPosition(int position) {
        this.position = position;
    }
}

File: app/src/main/java/com/example/bookworm/bottomMenu/feed/comments/Comment.kt
package com.example.bookworm.bottomMenu.feed.comments

import android.os.Parcelable
import com.example.bookworm.core.userdata.UserInfo
import com.google.firebase.firestore.Exclude
import kotlinx.parcelize.Parcelize

@Parcelize
data class Comment(
        var commentID: String = "", //댓글 ID
        var userToken: String ?= "", //댓글 작성자 토큰
        var contents: String ?= "", // 댓글 내용
        var madeDate: String ?= "", // 댓글 생성일자
        @Exclude
        var position: Int ?=0, //댓글의 위치
        @Exclude
        var duration: String? ="", //댓글 생성일자(~일전 등)
        @Exclude
        var creator: UserInfo?= null, // 댓글 작성자 정보
) : Parcelable //Content를 통해 전달할 수도 있을까봐 적용해둠
